src/net.cpp

```diff
@@ -2252,11 +2284,19 @@ bool CConnman::Start(CScheduler& scheduler, std::string& strNodeError, Options c
 
     SetBestHeight(connOptions.nBestHeight);
 
+    clientInterface = connOptions.uiInterface;
+
+    if (!InitBinds(connOptions.vBinds)) {
+        strError = _("Failed to listen on any port. Use -listen=0 if you want this.");
```

Comment:
This strError param is clunky (my fault!). We should do away with it and just use clientInterface->InitError() directly. That can be done as a follow-up though.